### ISSUE TITLE
Some drops unfarmable due to Twitch 'bug'

### ISSUE THREAD

**Issue by ArchOwlen** [DevilXD/TwitchDropsMiner#559]
As previously stated in issue/suggestion #557, some games can be correctly linked to Twitch, but Twitch does not consider it to be "Linked"

There are also games (such as Minecraft) that can not be linked, but still provide a redemption code for rewards (Minecraft's purple heart cape)
![image](https://github.com/user-attachments/assets/54a175e2-9adc-45fa-ac8e-1d93ba11e2aa)
![image](https://github.com/user-attachments/assets/718a96bc-278f-496f-bb0b-7af0ead2c01b)

As an update to #557, I was successfully able to re-link Blizzard games; However, Palworld was unable to be "Linked" despite being linked on the game's website when their drops were active

#557 was a suggestion on to bypass this Twitch "bug" and allow users to set a few (1-3) independent Priority games to be farmed despite this Twitch "bug" as there is no way to solve it outside of manually watching a stream

As per the wiki, it states that TDM is 'not intended for':
"Mining campaigns the managed account isn't linked to."

However, in my case, Palworld _is_ linked, but TDM doesn't support mining it
Thus, in turn, not being against the philosophy of "only games that are linked to the account"
![image](https://github.com/user-attachments/assets/8d3c23ac-7ebe-4839-b076-1ea8f5f073ce)
![image](https://github.com/user-attachments/assets/94c2a3bb-2f84-4b4e-a234-5491ab75dde5)

Which was claimed and is visible in my Steam inventory, despite what Twitch says:
![image](https://github.com/user-attachments/assets/1a3029e6-2ff8-4b29-8269-1b62439cf188)

So my suggestion was to add support for Priority games that fall under this Twitch "bug"; simply to allow TDM to accomplish its goal of 'Mining campaigns the managed account _is_ linked to'

Also looking at issue #84 (where I was referred to read before being locked out of issue #557), there is no mention of the concept of: "Twitch doesn't think X game is linked, despite it being properly linked on the game's website and still receiving drops" in the discussion; so this is a new concept being brought to the table

In short
- issue #557 is not listed under "features that will likely not be implemented"
- issue #557 does not go against the Project Goals, as the accounts are linked to the account
- issue #84 does not mention the idea of there being a "Twitch bug" and offering mining non-linked games as a solution

- Palworld can be successfully linked and provide drops, but TDM will not farm it due to being "not linked" on Twitch's side
- Minecraft (as another example) was a drop that could not be linked, and was simply a redemption code 

I hope this can be given a proper discussion this time around, rather than being locked immediately and being referred to unrelated goals/issues.
Thank you for your time.

**Comment by ArchOwlen** [comment]
Apologies for the number of edits, Github kept removing one of the screenshots if I added a new one

**Comment by DevilXD** [comment]
Hello,

> There are also games (such as Minecraft) that can not be linked, but still provide a redemption code for rewards (Minecraft's purple heart cape)

I am aware, they're called "Rewards". However, they're not considered drops by the miner, and thus aren't processed. This project focuses solely on timed drops that appear in the campaigns, Supporting rewards was never my goal, and wasn't requested by anyone so far, maybe minus some confused individuals who though that they're "drops". They're not, and they're far from them. I don't even know how hard would it be to support them, and given that there isn't really anyone interested, it's probably going to remain unsupported for at least quite a while, if not forever.

The miner isn't supposed to "replace you" in using the Twitch site: if you want the rewards, you'll need to get them the intended way.

> However, in my case, Palworld is linked, but TDM doesn't support mining it  
> Thus, in turn, not being against the philosophy of "only games that are linked to the account"

Unfortunately, you're not the one to decide and dictate what constitutes as "linked" here. According to Twitch, Palworld isn't linked - period. There's no further discussion here, and the miner isn't at fault here either. If you think this is a mistake and you want this fixed, you'll need to contact Twitch support and ask them to fix their linking service, so that it reports and displays correctly instead. Once the miner receives information that the account is properly linked, *then* it'll function as you'd expect. Otherwise, it won't. I won't be implementing any workarounds for this either, for reasons I've already pointed you to, that you've clearly didn't even read.

There's a good chance that, if you'd follow the clean reconnect process, it'd fix itself, as behavior like that has happened in the past, and is a result of using a stale session to perform the linking process. The accounts are "linked", but they really aren't, they may work, or they may not, the status is either reported correctly, or incorrectly. The best way to ensure proper linking is to follow the procedure. If that does not help, Twitch support is your only option.

**If any of your further messages is going to again argue that "your account is linked", while it clearly isn't, and you further refuse to acknowledge what I've said, you're going to be blocked from creating issues to this repository, and you will receive no further help from my side.** I'm pretty sure that my explanation is quite clear, and this isn't kindergarten. I won't be wasting any more of my free time for this bullshit. If you don't like it, then again, please use a different project. Or you can actually start listening to what I have to say.

**Comment by JourneyOver** [comment]
Just to randomly add to this discussion about stale sessions on the linking process and such, there actually was a bug on either twitches side or on palworlds side for the palworld drops (the 3 campaigns for palworld are over since the 11th so it's not an issue now) where literally no one was actually able to "link" their accounts on twitches side of things for all three campaigns (there was mulitple threads on palworlds subreddit about this) even though it was linked correctly on the palworld site for the drops and that they still gave the rewards for watching regardless of what it looked like on twitches site. 

But heres hoping whatever the issue was though with their campaigns is fixed in the future if they ever decide to do another round of twitch drops at some point, so people can link their accounts correctly.

**Comment by ArchOwlen** [comment]
> for reasons I've already pointed you to, that you've clearly didn't even read.

I have explicitly stated that:
- Nowhere in the Project Goals is the reasoning stated
- Nowhere in #84 is there reasoning behind "If Twitch refuses to properly connect"

Let's investigate #84 (Quotes by DevilXD)
>  it was really just a design choice I made at some point during development. Campaigns that aren't linked are simply not worth processing, since you can't claim their drops anyway

Design choices can be changed
They are worth processing _if you intend on claiming drops_

> There's no point in mining drops for a game you won't play, and if you are playing it, then you either already have your account linked, or you're about to do it because you want the drops anyway.

My account is linked, and I do intend on playing it, which is why I went out of my way to manually watch Palworld and watched Minecraft in the past

All other information is either
- Not stated by DevilXD
- Is stated by DevilXD in relation to other questions

If there is another source that I have not been linked to, please let me know.

Frankly, I wanted to support this project and try making my own pull requests, but this attitude coming from _the developer themselves_ towards a suggestion puts a bad taste in my mouth

**Comment by DevilXD** [comment]
> Nowhere in the Project Goals is the reasoning stated

It is, and if you've missed it, or you're just not getting it, then no amount of my explanation is going to work. I can't understand it for you, and thus I cannot help you here.

> I wanted to support this project and try making my own pull requests, but this attitude coming from the developer themselves towards a suggestion puts a bad taste in my mouth

Good. Please take your support somewhere else. I don't want to deal with people like you ever again.
